src/Qt-RSA/qrsaencryption.cpp

```diff
@@ -290,65 +289,117 @@ QByteArray decodeArray(const QByteArray &rawData, const QByteArray &privKey) {
+// --- static methods ---
+bool QRSAEncryption::generatePairKeyS(QByteArray &pubKey, QByteArray &privKey,
+                                      QRSAEncryption::Rsa rsa) {
 
-    return result;
+    return QRSAEncryption().generatePairKey(pubKey, privKey, rsa);
+}
+QByteArray QRSAEncryption::encodeS(const QByteArray &rawData, const QByteArray &pubKey,
+                                  QRSAEncryption::Rsa rsa)
+{
+    return QRSAEncryption().encode(rawData, pubKey, rsa);
 }
+QByteArray QRSAEncryption::decodeS(const QByteArray &rawData, const QByteArray &privKey,
+                                   QRSAEncryption::Rsa rsa)
+{
+    return QRSAEncryption().decode(rawData, privKey, rsa);
+}
+QByteArray QRSAEncryption::signMessageS(QByteArray rawData, const QByteArray &privKey,
+                                        Rsa rsa)
+{
+    return QRSAEncryption().signMessage(rawData, privKey, rsa);
+}
+bool QRSAEncryption::checkSignMessageS(const QByteArray &rawData, const QByteArray &pubKey,
+                                       Rsa rsa)
+{
+    return QRSAEncryption().checkSignMessage(rawData, pubKey, rsa);
+}
+// --- end of static methods ---
+
+bool QRSAEncryption::generatePairKey(QByteArray &pubKey, QByteArray &privKey, QRSAEncryption::Rsa rsa)
+{
+    int cnt{0};
+    bool keyGenRes{false};
+
+    do {
+        pubKey.clear();
+        privKey.clear();
+
+        switch (rsa) {
+
+            case RSA_64: {
+                if (!keyGenerator<int64_t>(pubKey, privKey)) {
+                    return false;
+                }
+                break;
+            }
+
+            case RSA_128: {
+                if (!keyGenerator<int128_t>(pubKey, privKey)) {
+                    return false;
+                }
+                break;
+            }
+        }
+
+    } while (!(keyGenRes = testKeyPair(pubKey, privKey, rsa)) && (++cnt < KEY_GEN_LIMIT));
 
-QRSAEncryption::QRSAEncryption() {
+    if(cnt >= KEY_GEN_LIMIT) qWarning() << QString("(Warning): Exceeded limit of key generation (%0)!").arg(KEY_GEN_LIMIT);
 
+    return (keyGenRes && cnt < KEY_GEN_LIMIT);
 }
 
-QByteArray QRSAEncryption::encode(const QByteArray &rawData, const QByteArray &pubKey) {
+// --- non-static methods ---
+QByteArray QRSAEncryption::encode(const QByteArray &rawData, const QByteArray &pubKey,
+                                  QRSAEncryption::Rsa rsa) {
 
-    switch (pubKey.size()) {
-        case RSA_64 / 4: {
+    switch (rsa) {
+        case RSA_64: {
             return encodeArray<uint64_t>(rawData, pubKey);
         }
 
-        case RSA_128 / 4: {
+        case RSA_128: {
             return encodeArray<uint128_t>(rawData, pubKey);
         }
-
-        default: return QByteArray();
     }
 }
+QByteArray QRSAEncryption::decode(const QByteArray &rawData, const QByteArray &privKey,
+                                  QRSAEncryption::Rsa rsa) {
 
-QByteArray QRSAEncryption::decode(const QByteArray &rawData, const QByteArray &privKey) {
```

Comment:
decode functions do not need the rsabits of the key, the size of key indicates the Rsabit